sktime/forecasting/base/adapters/_neuralforecast.py

```diff
@@ -179,6 +181,16 @@ def _fit(
         if not fh.is_all_out_of_sample(cutoff=self.cutoff):
             raise NotImplementedError("in-sample prediction is currently not supported")
 
+        if self.freq == "auto":
+            if fh.freq:
+                # interpret freq from ForecastingHorizon
+                self.freq = fh.freq
```

Comment:
This is not in line with sktime/sklearn principle that init attributes should not be modified afterwards. To handle this, you can initiate a new attribute in init, say `_freq` as `None` and here override with self.freq or fh.freq as applicable.

Reply:
> Also, is this a conventional good practice or there exists some guidelines that I can read to avoid such mistakes in future.

The extension templates, normally, though I noticed that the "do not mutate/overwrite" contract is not discussed very clearly. I added some clarifying explanations in this PR:

https://github.com/sktime/sktime/pull/6053

Reply:
> Are you suggesting

That suggestion would satisfy the contract - whether that's what @yarnabrina had in mind, of course only he can say (I'd guess: probably yes).

Reply:
Yes, I meant the same. You can do similar to how self.loss is handled when None is provided but MAE needs to be used etc.